src/librustdoc/visit_ast.rs

```diff
@@ -333,4 +363,102 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
             om.foreigns.push((item, renamed));
         }
     }
+
+    /// Run a few checks on the `#[doc]` attributes for an item:
+    /// - warn if a crate-level attribute is used on a non-`crate` item
+    /// - warn if `#[doc(inline)]`/`#[doc(no_inline)]` is used on a non-`use` item
+    /// - error if conflicting `#[doc(inline)]`/`#[doc(no_inline)]` are found
+    fn check_item_attrs(
+        &mut self,
+        span: Span,
+        def_id: DefId,
+        attrs: &'tcx [ast::Attribute],
+        item: AttrReceiverKind,
+    ) {
+        let mut specified_inline = None;
+
+        for (attr, (whole_attr_span, attr_style)) in attrs
+            .iter()
+            .filter(|attr| attr.has_name(sym::doc))
+            .filter_map(|attr| {
+                attr.meta_item_list()
+                    .map(|metas| metas.into_iter().zip(iter::repeat((attr.span, attr.style))))
+            })
+            .flatten()
+        {
+            // Check for crate-level attributes used on non-crate item
+            if item != AttrReceiverKind::Crate
+                && attr
+                    .meta_item()
+                    .map_or(false, |meta| CRATE_LEVEL_ATTRS.iter().any(|attr| meta.path == *attr))
+            {
+                let mut err = self.cx.sess().struct_span_warn(
+                    attr.meta_item().unwrap().span,
+                    "this attribute can only be applied at the crate level",
+                );
```

Comment:
Were you going to make this a lint so it can be `allow`ed or denied?

Reply:
Yes, still working on this ;)